src/core/jpx.js

```diff
@@ -137,2213 +62,14 @@ class JpxImage {
         const YOsiz = stream.getInt32() >>> 0; // Byte 16
         stream.skip(16);
         const Csiz = stream.getUint16(); // Byte 36
-        this.width = Xsiz - XOsiz;
-        this.height = Ysiz - YOsiz;
-        this.componentsCount = Csiz;
-        // Results are always returned as `Uint8ClampedArray`s.
-        this.bitsPerComponent = 8;
-        return;
+        properties.width = Xsiz - XOsiz;
+        properties.height = Ysiz - YOsiz;
+        properties.componentsCount = Csiz;
+        return properties;
```

Comment:
Just one more question, given the immediate return here: Can't we simply inline the object instead?
```suggestion
        return {
          width: Xsiz - XOsiz,
          height: Ysiz - YOsiz,
          // Results are always returned as `Uint8ClampedArray`s.
          bitsPerComponent: 8,
          componentsCount: Csiz,
        };
```